karma.js

```diff
@@ -16,62 +16,82 @@
 
 'use strict';
 
-var launchers = require('./launchers.json');
+module.exports = function configure(packageJSON, customWebpackConfig) {
 
-var config = {
-  plugins: [
-    require('karma-mocha'),
-    require('karma-phantomjs-launcher'),
-    require('karma-sauce-launcher'),
-    require('karma-browserify')
-  ],
-  basePath: '',
-  frameworks: [ 'browserify', 'mocha' ],
-  files: [
-    'lib/*.coffee',
-    'tests/*.spec.coffee'
-  ],
-  exclude: [],
-  preprocessors: {
-    '**/*.coffee': [ 'browserify' ]
-  },
-  browserify: {
-    debug: true,
-    transform: [ require('coffeeify') ],
-    extensions: [ '.js', '.coffee' ]
-  },
-  reporters: [ 'progress' ],
-  port: 9876,
-  colors: true,
-  autoWatch: false,
-  browsers: [ 'PhantomJS' ],
-  singleRun: true,
-  concurrency: Infinity,
+  var webpackConfig = customWebpackConfig || {
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          exclude: /(node_modules)/,
+          loader: "babel-loader",
+          query: {
+            presets: [ "env", "es2015" ]
+          }
+        }
+      ]
+    }
+  };
 
-  // Increase tolerance for unstable network connections
-  captureTimeout: 300000,
-  browserDisconnectTolerance: 5,
-  browserNoActivityTimeout: 300000
-};
+  var config = {
+    plugins: [
+      require('karma-mocha'),
+      require('karma-chrome-launcher'),
+      require('karma-sauce-launcher'),
+      require('karma-webpack')
+    ],
+    basePath: '',
+    frameworks: [ 'mocha' ],
+    files: [
+      'tests/**/*.spec.js'
```

Comment:
I actually think the tests should be written in JS and very rarely in TS (only when testing TS-specific things like class inheritance).